lib/api/bucketPutVersioning.js

```diff
@@ -116,10 +118,22 @@ function bucketPutVersioning(authInfo, request, log, callback) {
             return next(null, bucket, versioningConfiguration);
         }),
         (bucket, versioningConfiguration, next) => {
+            // check if replication is enabled if versioning is being suspended
+            const replicationConfig = bucket.getReplicationConfiguration();
+            const replicationEnabled = replicationConfig
+                && replicationConfig.rules
+                && replicationConfig.rules.some(r => r.enabled);
+            if (versioningConfiguration.Status === 'Suspended' &&
+                replicationEnabled) {
+                next(errors.InvalidBucketState
+                    .customizeDescription(invalidBucketStateMessage));
+                return;
+            }
             bucket.setVersioningConfiguration(versioningConfiguration);
             // TODO all metadata updates of bucket should be using CAS
             metadata.updateBucket(bucket.getName(), bucket, log, err =>
                 next(err, bucket));
+            return;
```

Comment:
not necessary as it is the last line of the function anyway

Reply:
true